### DISCUSSION TITLE
Mega-Bas Remote I/O Error

### DISCUSSION THREAD

**Discussion by drcooper345** [tagyoureit/nodejs-poolController#787] [App and O/S Config Questions] [answered]
Let me start by saying that @tagyoureit, @rstrouse, and others have done an amazing job putting together this incredible resource - so thank you.  
I have done hours of reading, lots of amazon shopping for parts, and I have put together a controller that should suit my system.  Sorry if this is info overload below but I wanted to get out details.

Pi 4
Sequent Building Automation Hat
Sequent 8 Relay Hat
Whitebox labs T3
IPS flow switch for pH and ORP
EZO pressure sensor
Air and water temp sensors

I am attempting to control my 25k gallon pool
Whisperflo VST
Pentair Ultratemp
Hayward Swim Pure Plus Chlorinator (Aquarite)
Pentair Muriatic Acid Feeder

I am working through getting things set up but have run into an issue that I can't seem to google my way out of.  In fact I have wiped my SD card and started from scratch more times than I care to admit.
My [Sequent Building Automation](https://sequentmicrosystems.com/collections/building-automation/products/building-automation-8-layer-stackable-hat-v4-for-raspberry-pi) megabas card does not seem to be fully talking to REM.  I have my whisperflo pump hooked up and it can be controlled via RS485 wired into the megabas.  I can run i2cdetect and see that the megabas card is there, and I can also see it in the i2c bus section of REM.  The problem is that the card is not responding to anything.  When running pm2 logs, any time I try to save a setting it throws an IO error.  

<img width="654" alt="image" src="https://github.com/tagyoureit/nodejs-poolController/assets/30439485/72b03d75-45cd-4246-9657-ed9573feb773">

I am not seeing temps from the probes I have set up, nor voltage or temp on the main page.  I am 99% certain I have the probes set up and talking correctly to nodejs.  

<img width="502" alt="image" src="https://github.com/tagyoureit/nodejs-poolController/assets/30439485/cb95dd8a-47bb-461e-814c-e42b01c9b814">

<img width="523" alt="image" src="https://github.com/tagyoureit/nodejs-poolController/assets/30439485/7cb4c417-f593-4a49-91fd-8c560aa76249">

<img width="362" alt="image" src="https://github.com/tagyoureit/nodejs-poolController/assets/30439485/1d61095c-bbe6-419a-9138-d6bed0e88268">

<img width="696" alt="image" src="https://github.com/tagyoureit/nodejs-poolController/assets/30439485/4b0860cf-575c-4d62-9326-709526acb4bd">

<img width="464" alt="image" src="https://github.com/tagyoureit/nodejs-poolController/assets/30439485/2ea5fc49-52b2-4e27-a49a-568d64147b22">

I am attaching the replay file from dashpanel and the console log from REM.  Any insight on how to fix this is greatly appreciated!
[consoleLog(2023-06-10_12-14-07).log](https://github.com/tagyoureit/nodejs-poolController/files/11712849/consoleLog.2023-06-10_12-14-07.log)
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/11712873/replay.zip)

**Comment by rstrouse** [comment, marked as answer]
Please update your firmware for the MEGA-BAS.  Also, shut down REM before you do this because the Sequent card will abort the firmware update as soon as anything tries to read any of the registers.  After you have updated you must also power down (not reboot) the pi and power it back up.

https://github.com/SequentMicrosystems/megabas-rpi/blob/master/update/README.md

**Reply by drcooper345** [reply]
That did the trick.  Thank you!

**Reply by rstrouse** [reply]
I'm almost as good as Google.  Just call me Giggle.

**Reply by drcooper345** [reply]
Well I appreciate it Giggles :) I have some plumbing to do tomorrow for the acid feeder and the flow cell, then I am going to start tinkering with the chlorinator.  Got a buck converter to put directly in the box to feed from the 24v power supply and convert down to 10v for the tap.   After that its trying to wrap my head around Home Assistant.  Thanks in advance  as I may be posting other questions in the coming days/weeks.
